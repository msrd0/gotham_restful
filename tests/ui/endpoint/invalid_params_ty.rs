#[macro_use]
extern crate gotham_restful;
use gotham_restful::gotham::hyper::Method;

#[derive(Resource)]
#[resource(endpoint)]
struct FooResource;

#[derive(Debug)]
struct FooParams {
	foo: String
}

#[endpoint(method = "Method::GET", uri = "", params = true)]
fn endpoint(_: FooParams) {
	unimplemented!()
}

fn main() {}
