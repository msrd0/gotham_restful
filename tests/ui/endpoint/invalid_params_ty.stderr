error[E0277]: the trait bound `FooParams: OpenapiType` is not satisfied
  --> $DIR/invalid_params_ty.rs:15:16
   |
15 | fn endpoint(_: FooParams) {
   |                ^^^^^^^^^ the trait `OpenapiType` is not implemented for `FooParams`
   |
  ::: $WORKSPACE/src/endpoint.rs
   |
   |     #[openapi_bound("Params: OpenapiType")]
   |                     --------------------- required by this bound in `gotham_restful::EndpointWithSchema::Params`

error[E0277]: the trait bound `FooParams: Clone` is not satisfied
  --> $DIR/invalid_params_ty.rs:15:16
   |
15 | fn endpoint(_: FooParams) {
   |                ^^^^^^^^^ the trait `Clone` is not implemented for `FooParams`
   |
  ::: $WORKSPACE/src/endpoint.rs
   |
   |     type Params: QueryStringExtractor<Body> + Clone + Sync;
   |                                               ----- required by this bound in `gotham_restful::EndpointWithSchema::Params`

error[E0277]: the trait bound `FooParams: StateData` is not satisfied
  --> $DIR/invalid_params_ty.rs:15:16
   |
15 | fn endpoint(_: FooParams) {
   |                ^^^^^^^^^ the trait `StateData` is not implemented for `FooParams`
   |
  ::: $WORKSPACE/src/endpoint.rs
   |
   |     type Params: QueryStringExtractor<Body> + Clone + Sync;
   |                  -------------------------- required by this bound in `gotham_restful::EndpointWithSchema::Params`

error[E0277]: the trait bound `FooParams: StaticResponseExtender` is not satisfied
  --> $DIR/invalid_params_ty.rs:15:16
   |
15 | fn endpoint(_: FooParams) {
   |                ^^^^^^^^^ the trait `StaticResponseExtender` is not implemented for `FooParams`
   |
  ::: $WORKSPACE/src/endpoint.rs
   |
   |     type Params: QueryStringExtractor<Body> + Clone + Sync;
   |                  -------------------------- required by this bound in `gotham_restful::EndpointWithSchema::Params`

error[E0277]: the trait bound `for<'de> FooParams: serde::de::Deserialize<'de>` is not satisfied
  --> $DIR/invalid_params_ty.rs:15:16
   |
15 | fn endpoint(_: FooParams) {
   |                ^^^^^^^^^ the trait `for<'de> serde::de::Deserialize<'de>` is not implemented for `FooParams`
   |
  ::: $WORKSPACE/src/endpoint.rs
   |
   |     type Params: QueryStringExtractor<Body> + Clone + Sync;
   |                  -------------------------- required by this bound in `gotham_restful::EndpointWithSchema::Params`
